core/src/main/resources/META-INF/services/org.apache.spark.deploy.history.EventFilterBuilder

```diff
@@ -0,0 +1 @@
+org.apache.spark.deploy.history.BasicEventFilterBuilder
```

Comment:
`EventFilterBuilder` is private in Spark. Do you mind if I ask why we use service loader?

Reply:
core module cannot refer SQL/core module which also has EventFilterBuilder.

Reply:
I see. I think that's possible via simply using reflection which I think is easier to read the codes. I think we're already doing this in few places such as `FileCommitProtocol.instantiate`

Seems a bit odds to use service loader for internal classes.

Reply:
https://github.com/apache/spark/blob/a5c7090ffafbfb4a8f312251b9ce8f25f71497cb/core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala#L1226-L1228

That is the way how FsHistoryProvider already deals with loading class in SQL core module.

Reply:
Okay, thanks. at least it's consistent.